lib/babe/state.go

```diff
@@ -15,6 +15,8 @@ import (
 	"github.com/ChainSafe/gossamer/lib/transaction"
 )
 
+//go:generate mockgen -source=./state.go -destination=./mock_state_test.go
```

Comment:
you don't need to use -source when the interface is in the same package, you can do instead
`go:generate mockgen -destination=./mock_state_test.go -package $GOPACKAGE . BlockState`
then it will generate using the `package babe`. However if you need to use the source flag you can use with this path format `github.com/ChainSafe/gossamer/babe`

Reply:
If I omit the ". BlockState" will it still generate mocks for all interfaces in the file? Or will I need to add line for each interface?

Reply:
You can have it with a comma like `. BlockState,SomeOtherInterface` if you want to have them on a single line.

I nowadays tend to write `//go:generate` comments above the test / test file needing it too since mocks are really a test only thing, so that usually removes the need to have all the go generate interfaces on a single line too. But that's just a preference, both work the same and it's unclear still which way is better :smile:

Reply:
okay cool I can make this change